### PR TITLE
Feature/resources remove warnings costura fody

### LINKED ISSUE #676: Costura Fody support
References are loaded fine, but I'm getting a lot of spam inside the console

```console
[12:34:50 WRN][] costura.autofac.dll.compressed does not contain assembly name in assembly: UnturnedGuard.OpenMod. <AssemblyName> and <RootNamespace> must be equal inside your plugins .csproj file
[12:34:50 WRN][] costura.autofac.pdb.compressed does not contain assembly name in assembly: UnturnedGuard.OpenMod. <AssemblyName> and <RootNamespace> must be equal inside your plugins .csproj file
[12:34:50 WRN][] costura.autofac.extensions.dependencyinjection.dll.compressed does not contain assembly name in assembly: UnturnedGuard.OpenMod. <AssemblyName> and <RootNamespace> must be equal inside your plugins .csproj file
[12:34:50 WRN][] costura.autofac.extensions.dependencyinjection.pdb.compressed does not contain assembly name in assembly: UnturnedGuard.OpenMod. <AssemblyName> and <RootNamespace> must be equal inside your plugins .csproj file
[12:34:50 WRN][] costura.contrib.bcl.ranges.dll.compressed does not contain assembly name in assembly: UnturnedGuard.OpenMod. <AssemblyName> and <RootNamespace> must be equal inside your plugins .csproj file
[12:34:50 WRN][] costura.costura.dll.compressed does not contain assembly name in assembly: UnturnedGuard.OpenMod. <AssemblyName> and <RootNamespace> must be equal inside your plugins .csproj file
[12:34:50 WRN][] costura.costura.pdb.compressed does not contain assembly name in assembly: UnturnedGuard.OpenMod. <AssemblyName> and <RootNamespace> must be equal inside your plugins .csproj file
[12:34:50 WRN][] costura.cronos.dll.compressed does not contain assembly name in assembly: UnturnedGuard.OpenMod. <AssemblyName> and <RootNamespace> must be equal inside your plugins .csproj file
[12:34:50 WRN][] costura.dnlib.dll.compressed does not contain assembly name in assembly: UnturnedGuard.OpenMod. <AssemblyName> and <RootNamespace> must be equal inside your plugins .csproj file
[12:34:50 WRN][] costura.dotnet.glob.dll.compressed does not contain assembly name in assembly: UnturnedGuard.OpenMod. <AssemblyName> and <RootNamespace> must be equal inside your plugins .csproj file
[12:34:50 WRN][] costura.inlineil.dll.compressed does not contain assembly name in assembly: UnturnedGuard.OpenMod. <AssemblyName> and <RootNamespace> must be equal inside your plugins .csproj file
[12:34:50 WRN][] costura.jetbrains.annotations.dll.compressed does not contain assembly name in assembly: UnturnedGuard.OpenMod. <AssemblyName> and <RootNamespace> must be equal inside your plugins .csproj file
[12:34:50 WRN][] costura.0harmony.dll.compressed does not contain assembly name in assembly: UnturnedGuard.OpenMod. <AssemblyName> and <RootNamespace> must be equal inside your plugins .csproj file
[12:34:50 WRN][] costura.litedb.dll.compressed does not contain assembly name in assembly: UnturnedGuard.OpenMod. <AssemblyName> and <RootNamespace> must be equal inside your plugins .csproj file
[12:34:50 WRN][] costura.microsoft.bcl.asyncinterfaces.dll.compressed does not contain assembly name in assembly: UnturnedGuard.OpenMod. <AssemblyName> and <RootNamespace> must be equal inside your plugins .csproj file
[12:34:50 WRN][] costura.microsoft.bcl.hashcode.dll.compressed does not contain assembly name in assembly: UnturnedGuard.OpenMod. <AssemblyName> and <RootNamespace> must be equal inside your plugins .csproj file
[12:34:50 WRN][] costura.microsoft.entityframeworkcore.dll.compressed does not contain assembly name in assembly: UnturnedGuard.OpenMod. <AssemblyName> and <RootNamespace> must be equal inside your plugins .csproj file
[12:34:50 WRN][] costura.microsoft.entityframeworkcore.abstractions.dll.compressed does not contain assembly name in assembly: UnturnedGuard.OpenMod. <AssemblyName> and <RootNamespace> must be equal inside your plugins .csproj file
[12:34:50 WRN][] costura.microsoft.entityframeworkcore.relational.dll.compressed does not contain assembly name in assembly: UnturnedGuard.OpenMod. <AssemblyName> and <RootNamespace> must be equal inside your plugins .csproj file
[12:34:50 WRN][] costura.microsoft.extensions.caching.abstractions.dll.compressed does not contain assembly name in assembly: UnturnedGuard.OpenMod. <AssemblyName> and <RootNamespace> must be equal inside your plugins .csproj file
[12:34:50 WRN][] costura.microsoft.extensions.caching.memory.dll.compressed does not contain assembly name in assembly: UnturnedGuard.OpenMod. <AssemblyName> and <RootNamespace> must be equal inside your plugins .csproj file
[12:34:50 WRN][] costura.microsoft.extensions.configuration.dll.compressed does not contain assembly name in assembly: UnturnedGuard.OpenMod. <AssemblyName> and <RootNamespace> must be equal inside your plugins .csproj file
[12:34:50 WRN][] costura.microsoft.extensions.configuration.abstractions.dll.compressed does not contain assembly name in assembly: UnturnedGuard.OpenMod. <AssemblyName> and <RootNamespace> must be equal inside 
```

As well it copied inside of my plugin directory all resources (i.e 235 files) :D

So, in this case, I would suggest making a such an extension or just put inside of the code, to ensure that the resource is costura to skip it right [here](https://github.com/openmod/openmod/blob/a0845362abdc5641375370cb72ab829589146fc5/framework/OpenMod.Core/Helpers/AssemblyHelper.cs#L41)
```cs
[SuppressMessage("ReSharper", "InvertIf")]
public static class ManifestResourceExtensions
{
    private const string CosturaResourceNameStart = "costura.";
    private const string CosturaResourceNameEnd = ".dll.compressed";
    private const int MinCosturaResourceNameCharactersLenght = 19;

    public static bool IsEmbeddedCosturaResource(this ManifestResource source)
    {
        if (Utf8String.IsNullOrEmpty(source.Name) == false)
        {
            if (source.IsEmbedded)
            {
                var name = source.Name.Value;
                if (name.Length > MinCosturaResourceNameCharactersLenght
                    && name.StartsWith(CosturaResourceNameStart)
                    && name.EndsWith(CosturaResourceNameEnd))
                {
                    return true;
                }
            }
        }
        return false;
    }
}
```

### PR DESCRIPTION
**Pull request by sunnamed434**
Closes #676

### CODE REVIEW

File: framework/OpenMod.Core/Helpers/AssemblyHelper.cs [hunk 1]
```diff
@@ -105,5 +115,19 @@
                 File.WriteAllText(filePath, fileContent);
             }
         }
+
+        private static bool IsCosturaResource(string? resourceName)
+        {
+            if (string.IsNullOrWhiteSpace(resourceName) == false)
```
[the remark this answers is not in the document]
Comment (sunnamed434) [reply framework/OpenMod.Core/Helpers/AssemblyHelper.cs]: Fixed

File: framework/tests/OpenMod.Core.Tests/Helpers/AssemblyHelperTests.cs [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+            // Act.
+            var result = IsCosturaResource(resourceName);
+
+            // Assert.
+            Assert.AreEqual(expectedResult, result);
+        }
+
+        private static bool IsCosturaResource(string resourceName)
+        {
+            if (string.IsNullOrWhiteSpace(resourceName) == false)
```
[the remark this answers is not in the document]
Comment (sunnamed434) [reply framework/tests/OpenMod.Core.Tests/Helpers/AssemblyHelperTests.cs]: Fixed

File: framework/OpenMod.Core/Helpers/AssemblyHelper.cs [hunk 3]
```diff
@@ -105,5 +115,19 @@
...
+            {
+                if (resourceName!.Length > c_MinCosturaResourceNameCharactersLenght
+                    && resourceName.StartsWith(c_CosturaResourceNameStart)
+                    && resourceName.EndsWith(c_CosturaResourceNameEnd))
+                {
+                    return true;
+                }
+            }
```
[the remark this answers is not in the document]
Comment (sunnamed434) [reply framework/OpenMod.Core/Helpers/AssemblyHelper.cs]: Fixed

File: framework/tests/OpenMod.Core.Tests/Helpers/AssemblyHelperTests.cs [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+            {
+                if (resourceName.Length > c_MinCosturaResourceNameCharactersLenght
+                    && resourceName.StartsWith(c_CosturaResourceNameStart)
+                    && resourceName.EndsWith(c_CosturaResourceNameEnd))
+                {
+                    return true;
+                }
+            }
```
[the remark this answers is not in the document]
Comment (sunnamed434) [reply framework/tests/OpenMod.Core.Tests/Helpers/AssemblyHelperTests.cs]: Fixed

File: framework/OpenMod.Core/Helpers/AssemblyHelper.cs (hunk 3 above)
[the remark this answers is not in the document]
Comment (sunnamed434) [reply framework/OpenMod.Core/Helpers/AssemblyHelper.cs]: Fixed

**Review by CyberAndrii — CHANGES_REQUESTED** [review]
I don't really like the idea of supporting a specific library. Instead it would be better to have interface `IOpenModResourceFilter` with method like `bool IsOpenModResource(string name)` that will be called via reflection (same way as `IServiceConfigurator`). In an implementation of that interface plugin devs will be able to specify what files they want to extract (e.g. configs) and which not. Default to extract all if no implementation is found in the assembly.

**Comment by sunnamed434** [comment]
> I don't really like the idea of supporting a specific library. Instead it would be better to have interface `IOpenModResourceFilter` with method like `bool IsOpenModResource(string name)` that will be called via reflection (same way as `IServiceConfigurator`). In an implementation of that interface plugin devs will be able to specify what files they want to extract (e.g. configs) and which not. Default to extract all if no implementation is found in the assembly.

Definitely cool idea

**Comment by Trojaner** [comment]
I agree with @CyberAndrii, closing for now

**Pull request closed without merging**